### PR TITLE
revert: add back phase0 bug fixes

### PR DESCRIPTION
**Pull request by vm**
Changes for after deploy today

### CODE REVIEW

**Review by JFrankfurt — APPROVED** [review]

File: src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx [hunk 1]
```diff
@@ -82,7 +82,8 @@ export default function MobileBalanceSummaryFooter({
   tokenAmount,
   nativeCurrencyAmount,
   isNative,
-}: BalanceSummaryProps) {
+  tokenAddress,
+}: BalanceSummaryProps & { tokenAddress?: string }) {
```
Comment (JFrankfurt) [inline src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx:86]: you should already have `tokenAddress` available on `tokenAmount`

**Review by JFrankfurt — CHANGES_REQUESTED** [review]

File: src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx (hunk 1 above)
Comment (vm) [reply src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx]: good pt, fixed

Comment (JFrankfurt) [reply src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx]: image

it appears to be working in the case I just tested... 🤔

Comment (JFrankfurt) [reply src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx]: image

I've tested: (on ethereum w/ a connected wallet) ETH ("NATIVE"), WETH, USDT (no balance), DAI (w/ balance) and all worked as expected

Comment (vm) [reply src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx]: hmmmmm

File: src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx [hunk 2]
```diff
@@ -123,7 +121,7 @@ export default function MobileBalanceSummaryFooter({
           </BalanceTotal>
         </BalanceInfo>
       )}
-      <SwapButton to={`/swap?outputCurrency=${outputTokenAddress}`}>
+      <SwapButton to={`/swap?outputCurrency=${tokenAmount?.currency.address}`}>
```
Comment (JFrankfurt) [inline src/components/Tokens/TokenDetails/MobileBalanceSummaryFooter.tsx:124]: I feel like we shouldn't leave the option open for a swap button rendered w/o a valid url. Could do either: 
- don't render w/o a valid url including outputCurrency
- render, but with only `/swap` and no query parameters

**Comment by vm** [comment]
the issue is the tokenamount should be created with 0 when there is no wallet connected potentially? i am not sure if that's the best solution bc to me 0 balance and no balance are different.

**Pull request merged**